### PR TITLE
Remove unused `test_iter`

### PR DESCRIPTION
**Pull request by goldmermaid**
The `test_iter` is never used after being defined.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1046/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1046/

**Comment by AnirudhDagar** [comment]
@astonzhang Whenever this is merged, I'll send a small fix for the broken training loss in PyTorch of the same notebook. Waiting for this to be reviewed and merged. Otherwise, my PR may create merge conflicts.
